packages/jest-util/src/stacktrace.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+// filter for noisy stack trace lines
+const FILTER_NOISE =
+  /(\/|\\)jest(-.*?)?(\/|\\)(vendor|build|node_modules|packages)(\/|\\)/;
+
+const toDiscard = (filename: string, index: number): boolean => {
+  return index !== 0 && FILTER_NOISE.test(filename);
+};
+
+const setPrepareStackTrace = (Error: Class<Error>) => {
+  Error.prepareStackTrace = (error, stacktrace) => {
+    const filteredStacktrace = stacktrace.filter(
+      (callsite, index) => !toDiscard(callsite.getFileName() || '', index)
+    ).map(callsite => {
+      const functionName = callsite.getFunctionName() || '<anonymous>';
+      const filename = callsite.getFileName() || '';
+      const line = callsite.getLineNumber() || '0';
+      const column = callsite.getColumnNumber() || '0';
+      return `at ${functionName} (${filename}:${line}:${column})`;
```

Comment:
yeah, it seems like this is missing a bunch of stuff. I've pointed to the documentation multiple times now, here is what CallSite provides:

```
getThis: returns the value of this
getTypeName: returns the type of this as a string. This is the name of the function stored in the constructor field of this, if available, otherwise the object's [[Class]] internal property.
getFunction: returns the current function
getFunctionName: returns the name of the current function, typically its name property. If a name property is not available an attempt will be made to try to infer a name from the function's context.
getMethodName: returns the name of the property of this or one of its prototypes that holds the current function
getFileName: if this function was defined in a script returns the name of the script
getLineNumber: if this function was defined in a script returns the current line number
getColumnNumber: if this function was defined in a script returns the current column number
getEvalOrigin: if this function was created using a call to eval returns a CallSite object representing the location where eval was called
isToplevel: is this a toplevel invocation, that is, is this the global object?
isEval: does this call take place in code defined by a call to eval?
isNative: is this call in native V8 code?
isConstructor: is this a constructor call?
```

Please make sure to write enough tests so that this new way of generating stack traces doesn't lose any data over the default printing. You may want to take a look at how v8 creates stack traces and work backwards from that.